src/lxc/confile.c

```diff
@@ -1217,6 +1219,53 @@ static int set_config_init_gid(const char *key, const char *value,
 	return 0;
 }
 
+static int set_config_init_groups(const char *key, const char *value,
+				  struct lxc_conf *lxc_conf, void *data)
+{
```

Comment:
This function should make use of our cleanup macros and not rely on manual freeing:
```c
static int set_config_init_groups(const char *key, const char *value,
                                  struct lxc_conf *lxc_conf, void *data)
{
        __do_free char *dup = NULL;
        __do_free gid_t *init_groups = NULL;
        char *token;
        int num_groups = 0;
        int iter = 0;

        if (lxc_config_value_empty(value))
                return clr_config_init_groups(key, lxc_conf, NULL);

        dup = strdup(value);
        if (!dup)
                return -1;

        lxc_iterate_parts(token, dup, " \t") num_groups++;

        if (num_groups == 0)
                return clr_config_init_groups(key, lxc_conf, NULL);

        init_groups = malloc(sizeof(gid_t) * num_groups);
        if (!init_groups)
                return -1;

        strcpy(dup, value);
        lxc_iterate_parts(token, dup, " \t") {
                gid_t group;
                if (lxc_safe_uint(token, &group) < 0)
                        return -1;
                init_groups[iter++] = group;
        }

        lxc_conf->init_groups.size = num_groups;
        lxc_conf->init_groups.list = move_ptr(init_groups);

        return 0;
}
```

Reply:
Ok, good to know. I'll make myself comfortable with `__do_free` and `move_ptr`. Less is more - but any change we can keep  the`value_dup` instead of `dup` to not confuse the syntax highlighting here :D

Reply:
Done.